vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinSavedRequestAwareAuthenticationSuccessHandler.java

```diff
@@ -78,8 +79,16 @@
         @Override
         public void sendRedirect(HttpServletRequest request,
                 HttpServletResponse response, String url) throws IOException {
+            String redirectUrl;
+            if (response.getHeader(SAVED_URL_HEADER) != null) {
+                redirectUrl = response.getHeader(SAVED_URL_HEADER);
+            } else {
+                redirectUrl = url;
+            }
```

Comment:
Same as https://github.com/vaadin/spring/pull/780#discussion_r628306150

Reply:
This was fixed in the other PR